packages/menu-bar/src/vaadin-menu-bar-interactions-mixin.js

```diff
@@ -144,38 +145,75 @@ export const InteractionsMixin = (superClass) =>
 
     /**
      * @param {!KeyboardEvent} event
-     * @protected
+     * @private
      */
-    _onKeydown(event) {
+    _onArrowDown(event) {
+      // Prevent page scroll.
+      event.preventDefault();
+
       const button = this._getButtonFromEvent(event);
-      if (button) {
-        if (event.key === 'Escape') {
-          this._hideTooltip();
-        }
+      if (button === this._expandedButton) {
+        // Menu opened previously, focus first item
+        this._focusFirstItem();
+      } else {
+        this.__openSubMenu(button, event);
+      }
+    }
+
+    /**
+     * @param {!KeyboardEvent} event
+     * @private
+     */
+    _onArrowUp(event) {
+      // Prevent page scroll.
+      event.preventDefault();
+
+      const button = this._getButtonFromEvent(event);
+      if (button === this._expandedButton) {
+        // Menu opened previously, focus last item
+        this._focusLastItem();
+      } else {
+        this.__openSubMenu(button, event, { focusLast: true });
+      }
+    }
+
+    /**
+     * Override an event listener from `KeyboardMixin`:
+     * - to close the sub-menu for expanded button,
+     * - to close a tooltip for collapsed button.
+     *
+     * @param {!KeyboardEvent} event
+     * @protected
+     * @override
+     */
+    _onEscape(event) {
+      if (event.composedPath().includes(this._expandedButton)) {
+        this._close(true);
+      } else {
+        this._hideTooltip();
+      }
+    }
+
+    /**
+     * Override an event listener from `KeyboardMixin`.
+     *
+     * @param {!KeyboardEvent} event
+     * @protected
+     * @override
+     */
+    _onKeyDown(event) {
+      switch (event.key) {
+        case 'ArrowDown':
+          this._onArrowDown(event);
+          break;
+        case 'ArrowUp':
+          this._onArrowUp(event);
+          break;
+        default:
+          super._onKeyDown(event);
```

Comment:
Just to make sure, what is the idea behind calling the super method only in the default case?

Reply:
The idea is to enable `_onEscape` key handler provided by `KeyboardMixin`.